controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go

```diff
@@ -191,6 +192,7 @@ func (in *KubeadmControlPlane) ValidateUpdate(old runtime.Object) error {
 	allErrs = append(allErrs, in.validateVersion(prev.Spec.Version)...)
 	allErrs = append(allErrs, validateEtcd(&in.Spec, &prev.Spec)...)
```

Comment:
Note: Per default the ignition field will be immutable in KCP. If that is not intended you have to add it to `allowedPaths` in l.122++ (and a corresponding test case)

Reply:
Afaik the nodes are not getting replaced, but the config will be used for new nodes.

Reply:
The problem is only if the fields are immutable you can't change them (without disabling the webhook) during the lifetime of the cluster, which is probably why we made all cloud-init (not sure if all but a lot of them) related fields mutable

Reply:
>@sbueringer what happens to the control plane nodes when updating a KCP? Do they get replaced?

Turns out that yes, when updating the KCP a rolling update gets triggered and the control plane machine(s) get gracefully replaced. Therefore, we definitely want to make Ignition mutable. Thanks for the great point of feedback!

Reply:
Ignition fields are now mutable in KCP. This can be resolved.